src/Microsoft.DotNet.XHarness.iOS.Shared/CrashSnapshotReporter.cs

```diff
@@ -0,0 +1,202 @@
+            this.isDevice = isDevice;
+            this.deviceName = deviceName;
+            this.tempFileProvider = tempFileProvider ?? Path.GetTempFileName;
+
+            symbolicateCrashPath = Path.Combine(processManager.XcodeRoot, "Contents", "SharedFrameworks", "DTDeviceKitBase.framework", "Versions", "A", "Resources", "symbolicatecrash");
+            if (!File.Exists(symbolicateCrashPath))
+                symbolicateCrashPath = Path.Combine(processManager.XcodeRoot, "Contents", "SharedFrameworks", "DVTFoundation.framework", "Versions", "A", "Resources", "symbolicatecrash");
+            if (!File.Exists(symbolicateCrashPath))
+                symbolicateCrashPath = null;
+        }
+
+        public async Task StartCaptureAsync()
+        {
+            initialCrashes = await CreateCrashReportsSnapshotAsync();
+        }
+
+        public async Task EndCaptureAsync(TimeSpan timeout)
+        {
+            // Check for crash reports
+            var stopwatch = Stopwatch.StartNew();
+
+            do
+            {
+                var newCrashFiles = await CreateCrashReportsSnapshotAsync();
+                newCrashFiles.ExceptWith(initialCrashes);
+
+                if (newCrashFiles.Count == 0)
+                {
+                    if (stopwatch.Elapsed.TotalSeconds > timeout.TotalSeconds)
+                    {
+                        break;
+                    }
+                    else
+                    {
+                        log.WriteLine(
+                            "No crash reports, waiting a second to see if the crash report service just didn't complete in time ({0})",
+                            (int)(timeout.TotalSeconds - stopwatch.Elapsed.TotalSeconds));
+
+                        Thread.Sleep(TimeSpan.FromSeconds(1));
+                    }
+
+                    continue;
+                }
+
+                log.WriteLine("Found {0} new crash report(s)", newCrashFiles.Count);
+
+                IEnumerable<ILog> crashReports;
+                if (!isDevice)
+                {
+                    crashReports = new List<ILog>(newCrashFiles.Count);
+                    foreach (var path in newCrashFiles)
+                    {
+                        logs.AddFile(path, $"Crash report: {Path.GetFileName(path)}");
+                    }
+                }
+                else
+                {
+                    // Download crash reports from the device. We put them in the project directory so that they're automatically deleted on wrench
+                    // (if we put them in /tmp, they'd never be deleted).
+                    crashReports = newCrashFiles
+                        .Select(async crash => await ProcessCrash(crash))
+                        .Select(t => t.Result)
+                        .Where(c => c != null);
+                }
+
+                foreach (var cp in crashReports)
+                {
+                    WrenchLog.WriteLine("AddFile: {0}", cp.FullPath);
```

Comment:
might be a good idea to get rid of all the references to Wrench, it moved to the CI graveyard a few years ago 😄

Reply:
trust me, I want to, I need to find out how :/

Reply:
There's not many and from what I have seen the logging is turned off there as well, so we could be safe to remove it. Can happen later